### PR TITLE
http: 100-continue is case insensitive in Expect header

### LINKED ISSUE #20602: 100-continue should be case insensitive per rfc 7231
[5.1.1](https://datatracker.ietf.org/doc/html/rfc7231#section-5.1.1).  Expect

   The "Expect" header field in a request indicates a certain set of
   behaviors (expectations) that need to be supported by the server in
   order to properly handle this request.  The only such expectation
   defined by this specification is 100-continue.

     Expect  = "100-continue"

   The Expect field-value is case-insensitive.

### PR DESCRIPTION
**Pull request by lambdai**
Commit Message:
Fixed a bug where ``100-continue`` comparison in the ``Expect`` header field was case sensitive.

Signed-off-by: Yuchen Dai <silentdai@gmail.com>

Additional Description:
Risk Level: LOW
Testing: modified the "100-continue" string literal containing both upper case and lower case
Docs Changes:
Release Notes: 
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA] Fix #20602
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Review by daixiang0 — DISMISSED** [review]
Good catch, lgtm!

**Review by adisuissa — DISMISSED** [review]
LGTM, thanks!
/assign @alyssawilk

File: docs/root/version_history/current.rst [hunk 1]
```diff
@@ -54,7 +54,7 @@ Bug Fixes
 * data plane: fixing error handling where writing to a socket failed while under the stack of processing. This should genreally affect HTTP/3. This behavioral change can be reverted by setting ``envoy.reloadable_features.allow_upstream_inline_write`` to false.
 * eds: fix the eds cluster update by allowing update on the locality of the cluster endpoints. This behavioral change can be temporarily reverted by setting runtime guard ``envoy.reloadable_features.support_locality_update_on_eds_cluster_endpoints`` to false.
 * http: fixed a bug where %RESPONSE_CODE_DETAILS% was not set correctly in :ref:`request_headers_to_add <envoy_v3_api_field_config.route.v3.RouteConfiguration.request_headers_to_add>`.
-* http: fixed a bug where ``100-continue`` comparison in the ``Expect`` header field was case sensitive.
+* http: fixed a bug where ``100-continue`` comparison in the ``Expect`` request header field was case sensitive. This RFC compliance behavior can be temporarily reverted by setting runtime guard ``envoy.reloadable_features.http_100_continue_case_insensitive`` to false.
```
[the remark this answers is not in the document]
Comment (lambdai) [reply docs/root/version_history/current.rst]: done. Thanks!

**Review by mattklein123 — APPROVED** [review]
Thanks!

**Pull request merged**